Support Spark 3.0

**Describe the bug**
```
./bin/spark-submit --master spark://KANGTIAN-MB0:7077 --class org.apache.spark.examples.SparkPi     --repositories https://raw.github.com/banzaicloud/spark-metrics/master/maven-repo/releases     --packages com.banzaicloud:spark-metrics_2.11:2.3-2.1.0,io.prometheus:simpleclient:0.3.0,io.prometheus:simpleclient_dropwizard:0.3.0,io.prometheus:simpleclient_pushgateway:0.3.0,io.dropwizard.metrics:metrics-core:3.1.2     ~/Documents/program/spark/examples/jars/spark-examples_2.11-2.4.4.jar  1000
```

ERROR:
```
19/11/08 13:22:47 INFO SparkContext: Successfully stopped SparkContext
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/spark/internal/Logging$class
	at com.banzaicloud.spark.metrics.sink.PrometheusSink.<init>(PrometheusSink.scala:47)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.spark.metrics.MetricsSystem.$anonfun$registerSinks$1(MetricsSystem.scala:216)
	at scala.collection.mutable.HashMap.$anonfun$foreach$1(HashMap.scala:149)
	at scala.collection.mutable.HashTable.foreachEntry(HashTable.scala:237)
	at scala.collection.mutable.HashTable.foreachEntry$(HashTable.scala:230)
	at scala.collection.mutable.HashMap.foreachEntry(HashMap.scala:44)
	at scala.collection.mutable.HashMap.foreach(HashMap.scala:149)
	at org.apache.spark.metrics.MetricsSystem.registerSinks(MetricsSystem.scala:196)
	at org.apache.spark.metrics.MetricsSystem.start(MetricsSystem.scala:106)
	at org.apache.spark.SparkContext.<init>(SparkContext.scala:571)
	at org.apache.spark.SparkContext$.getOrCreate(SparkContext.scala:2555)
	at org.apache.spark.sql.SparkSession$Builder.$anonfun$getOrCreate$5(SparkSession.scala:896)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.SparkSession$Builder.getOrCreate(SparkSession.scala:887)
	at org.apache.spark.examples.SparkPi$.main(SparkPi.scala:31)
	at org.apache.spark.examples.SparkPi.main(SparkPi.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:901)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:179)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:202)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:89)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:980)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:989)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.lang.ClassNotFoundException: org.apache.spark.internal.Logging$class
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 32 more
```

**Steps to reproduce the issue:**
<!-- Please describe the steps to reproduce the issue. -->

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

**Screenshots**
![image](https://user-images.githubusercontent.com/39302402/68451861-75b70780-022b-11ea-967a-dafec1541fbd.png)


**Additional context**
<!-- Add any other context about the problem like release numberm version, branch, etc. -->

--- a/build.sbt
+++ b/build.sbt
@@ -1,6 +1,5 @@
-lazy val scala212 = "2.12.10"
-lazy val scala211 = "2.11.12"
-lazy val supportedScalaVersions = List(scala212, scala211)
+lazy val scala212 = "2.12.12"
+lazy val supportedScalaVersions = List(scala212)
 
 lazy val root = (project in file("."))
   .settings(
@@ -17,18 +16,18 @@ lazy val root = (project in file("."))
     licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0")),
     scalaVersion := scala212,
     crossScalaVersions := supportedScalaVersions,
-    version      := "2.4-1.0.6",
+    version      := "3.0-1.1.0",
     libraryDependencies ++= Seq(
-      "io.prometheus" % "simpleclient" % "0.8.1",
-      "io.prometheus" % "simpleclient_dropwizard" % "0.8.1",
-      "io.prometheus" % "simpleclient_pushgateway" % "0.8.1",
-      "io.dropwizard.metrics" % "metrics-core" % "3.1.5",
-      "io.prometheus.jmx" % "collector" % "0.12.0",
-      "org.apache.spark" %% "spark-core" % "2.4.4" % Provided,
+      "io.prometheus" % "simpleclient" % "0.9.0",
+      "io.prometheus" % "simpleclient_dropwizard" % "0.9.0",
+      "io.prometheus" % "simpleclient_pushgateway" % "0.9.0",
+      "io.dropwizard.metrics" % "metrics-core" % "4.1.1" % Provided,
+      "io.prometheus.jmx" % "collector" % "0.14.0",
+      "org.apache.spark" %% "spark-core" % "3.0.1" % Provided,
       "com.novocode" % "junit-interface" % "0.11" % Test,
-      // Spark shaded jetty is not resolved in scala 2.11
-      // Described in https://issues.apache.org/jira/browse/SPARK-18162?focusedCommentId=15818123#comment-15818123
-      "org.eclipse.jetty" % "jetty-servlet"  % "9.4.18.v20190429" % Test
+//      // Spark shaded jetty is not resolved in scala 2.11
+//      // Described in https://issues.apache.org/jira/browse/SPARK-18162?focusedCommentId=15818123#comment-15818123
+//      "org.eclipse.jetty" % "jetty-servlet"  % "9.4.18.v20190429" % Test
     ),
     testOptions in Test := Seq(Tests.Argument(TestFrameworks.JUnit, "-a"))
   )

